backend/ethereum/subscription/eventsub.go

```diff
@@ -109,6 +109,8 @@ func calcStartBlock(ctx context.Context, chain ethereum.ChainReader, pastBlocks
 // Read reads all past and future events into `sink`.
 // Can be aborted by cancelling `ctx` or `Close()`.
 // It is possible that the same event is read more than once.
+// After casting the generic event to a specific type, the `Raw` log field
+// will be nil. Use the `Log` field of the generic event instead.
```

Comment:
Why does the Raw log field become nil? Is this intentional?

Reply:
This is not ideal, but i dont think it is possible to do better without the use of *reflections*.
`UnpackLog` does not set the `Raw` field of the `Event.Data` that it unpacks into.  
Since i do not know the exact type of `Event.Data`, the `Raw` field remains `nil`.  
This a bit of a trade-off between generic and concrete event subs.